### PR TITLE
fix(setupchecks): Add setup checks for current checks

### PR DESCRIPTION
**Pull request by nickvergessen**
grafik

### Todo

- [ ] Followup: Remove old checks within the talk settings?

### 🏁 Checklist

- [x] ⛑️ Tests (unit and/or integration) are included or not possible
- [x] 📘 API documentation in `docs/` has been updated or is not required
- [x] 🔖 Capability is added or not needed

### CODE REVIEW

File: lib/SetupCheck/FederationLockCache.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	}
+
+	public function run(): SetupResult {
+		if (!$this->talkConfig->isFederationEnabled()) {
+			return SetupResult::success();
+		}
+		if (!$this->cacheFactory->createLocking('talkroom_') instanceof NullCache) {
+			return SetupResult::success();
+		}
+		return SetupResult::warning($this->l->t('It is heavily recommended to configure "memcache.locking" when Talk Federation is enabled.'));
```
Comment (SystemKeeper) [inline lib/SetupCheck/FederationLockCache.php:41]: Would "highly recommended" be better here?

File: lib/SetupCheck/RecommendCache.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	}
+
+	public function run(): SetupResult {
+		if ($this->talkConfig->getSignalingMode() === Config::SIGNALING_INTERNAL) {
+			return SetupResult::success();
+		}
+		if ($this->cacheFactory->isAvailable()) {
+			return SetupResult::success();
+		}
+		return SetupResult::warning($this->l->t('It is heavily recommended to configure a memory cache when running Nextcloud Talk with a High-performance backend.'),);
```
Comment (SystemKeeper) [inline lib/SetupCheck/RecommendCache.php:43]: Same question. Also could we link to https://docs.nextcloud.com/server/latest/admin_manual/configuration_server/caching_configuration.html ?

Comment (nickvergessen) [reply lib/SetupCheck/RecommendCache.php:43]: I tried it, but it doesn't have a go.php redirect, have to do that tomorrow

Comment (nickvergessen) [reply lib/SetupCheck/RecommendCache.php]: Done in https://github.com/nextcloud/documentation/pull/12137

**Review by SystemKeeper — APPROVED** [review]

**Pull request merged**